amplfi/train/data/datasets/testing/mdc.py

```diff
@@ -0,0 +1,122 @@
+    mask = phi > np.pi
+    phi[mask] -= 2 * np.pi
+
+    return phi
+
+
+class MDCDataset(FlowDataset):
+    """
+    Testing dataset for iterating over MDC data
+    """
+
+    def __init__(self, mdc_file: Path, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mdc_file = mdc_file
+        self.i = 0
+
+    def test_dataloader(self) -> torch.utils.data.DataLoader:
+        # load in the strain data and parameters
+        strain = []
+        parameters = {}
+        with h5py.File(self.mdc_file, "r") as f:
+            for ifo in self.hparams.ifos:
+                strain.append(torch.tensor(f[ifo][:], dtype=torch.float32))
+
+            for parameter in self.hparams.inference_params + ["ra", "gpstime"]:
+                # skip phi since the injeciton file contains ra and gpstime
+                if parameter == "phi":
+                    continue
+                parameters[parameter] = torch.tensor(
+                    f[parameter][:], dtype=torch.float32
+                )
+
+        parameters["phi"] = torch.tensor(
+            phi_from_ra(
+                parameters["ra"].numpy(), parameters["gpstime"].numpy()
+            )
+        )
+        strain = torch.stack(strain, dim=1)
+
+        # based on psd length, fduration and kernel length, and padding,
+        # determine slice indices. It is assumed the coalescence
+        # time of the waveform is in the middle
+        middle = strain.shape[-1] // 2
+        post = (
+            self.waveform_sampler.padding
+            + self.waveform_sampler.ringdown_duration
+        )
+        pre = (
+            post
+            - self.hparams.kernel_length
+            - (self.hparams.fduration // 2)
+            - self.hparams.psd_length
+        )
+
+        post = int(post * self.hparams.sample_rate)
+        pre = int(pre * self.hparams.sample_rate)
+
+        start, stop = middle + pre, middle + post
+        strain = strain[..., start:stop]
+
+        # convert parameters to a tensor
+        parameters = [
+            torch.Tensor(parameters[k]) for k in self.hparams.inference_params
+        ]
+        parameters = torch.vstack(parameters).T
+
+        # build dataset
+        dataset = torch.utils.data.TensorDataset(strain, parameters)
+
+        return torch.utils.data.DataLoader(
+            dataset, batch_size=1, num_workers=12
+        )
+
+    def on_after_batch_transfer(self, batch, _):
+        if not self.trainer.testing:
+            raise ValueError("Use of the MDC dataset is for testing only")
+
+        X, parameters = batch
+
+        X, psds = self.psd_estimator(X)
+        X = self.whitener(X, psds)
+
+        # scale parameters
+        parameters = self.scale(parameters)
+
+        # calculate asds
+        freqs = torch.fft.rfftfreq(X.shape[-1], d=1 / self.hparams.sample_rate)
+        num_freqs = len(freqs)
+        psds = torch.nn.functional.interpolate(
+            psds, size=(num_freqs,), mode="linear"
+        )
```

Comment:
what is this for?

Reply:
this makes sure the psds are the same length of the data so they can be concatenated before passing to the embedding. (This is done under the hood in the `Whitener`)

Reply:
Let's put that down as a one-line comment?